### PR TITLE
Docs: align sqlite3 docs with versionadded/versionchanged recommendations

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
📚 Documentation preview 📚: https://cpython-previews--114400.org.readthedocs.build/

### CODE REVIEW

**Review by CAM-Gerlach — COMMENTED** [review]
LGTM overall, thanks.

To note, auditing the rest of the annotations, there's also a related issue: On lines 356 766, 825, 994, 1013 and 1041 the `versionchanged:: 3.13` for the keyword argument deprecations should probably be a `deprecated-removed:: 3.13 3.15`, since deprecations should and generally are documented as such (despite the imprecision of not currently being able to distinguish deprecations.removals of objects versus parameters and behaviors, which I'm working on fixing).

You might want to fix that either here, or in another PR.

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -343,17 +343,17 @@ Module functions
    .. audit-event:: sqlite3.connect database sqlite3.connect
    .. audit-event:: sqlite3.connect/handle connection_handle sqlite3.connect
 
-   .. versionadded:: 3.4
-      The *uri* parameter.
+   .. verisonchanged:: 3.4
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
   .. versionchanged:: 3.4
```

Fix typo in directive name (causing docs build to fail)

**Comment by erlend-aasland** [comment]
> To note, auditing the rest of the annotations, there's also a related issue: On lines 356 766, 825, 994, 1013 and 1041 the `versionchanged:: 3.13` for the keyword argument deprecations should probably be a `deprecated-removed:: 3.13 3.15`, since deprecations should and generally are documented as such (despite the imprecision of not currently being able to distinguish deprecations.removals of objects versus parameters and behaviors, which I'm working on fixing).

I'm not so sure about these. IMO, we're talking about behaviour changes (params becoming positional-only or keyword-only), not _removals_ of parameters.

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]: Thanks!

**Comment by CAM-Gerlach** [comment]
> I'm not so sure about these. IMO, we're talking about behaviour changes (params becoming positional-only or keyword-only), not removals of parameters.

Right, thought just to note, elsewhere `deprecated-removed` is routinely used for behavior deprecations too, as there isn't currently is a finer grained directive to distinguish deprecations of entire objects/APIs from deprecations of specific behaviors/functionality of the object (which is something I'm working on enhancing our existing custom `deprecated-removed` directive to allow).

> IOW, I'd like to pass on this suggestion for this PR. We can amend them in a follow-up PR if needed.

Yup, entirely fair of course :+1:

**Pull request merged**